space-framework/docs/space-framework.md

```diff
@@ -0,0 +1,92 @@
+3. Unmanaged Space: The SF supports discovery of existing pSpaces on a space provider and adding them as "Unmanaged Spaces" into the framework. This mode is similar to the import mode in terms of the "source of truth" - the desired state is defined by the pSpace. However, as oppose to import, here the pSpaces are automatically discovered through the space provider regular interface and the pSpace access information is retrieved from teh space provider similary to how it is done in the managed space case. The SF supports pattern based filtering rules to limit the discovery of pSpaces. 
+
+### Space Manager 
+The Space Manager (SM) is a Kuberentes controller that is responsible for maintaing the state of the Space objects. The SM reconcile the Space and SpaceProviderDesc objects, and monitor the state of the pSpaces through the space provider.   
+The SM uses a library of space provider adaptors to communicate with the space providers. Currently the SF includes 3 space provider adaptors KCP, KIND, and KubeFlex.
+
+### Space provider adaptors
+The space manager uses a set of space provider adaptors that interacts with the space provider. All space provider adaptors implement a simple "space client interface" that includes basic pSpace life cycle operatios. In addition the interface also defines the events that sent from the adaptors to the space manager.
+
+### Space aware client (SAC)
+The Space aware client (msclient) allows clients/controllers to easily get access to the underlying space by simply using the space name. There is no direct interaction between the msclient and the Space  provider, and therefore the msclient is transparent to the specific provider of the spaces. 
+![Space Aware Client](SAC.drawio.svg "Space Aware Client")
+The only requirement is that the space can be accessed through regular Kube APIs when using the appropriate kubeconfig information. 
+
+**Main features**  
+- Constantly watch for changes in the available spaces. The SAC is using an informer on Space objecs using the SF Space client-go APIs.
+    - Holds a cache of the pSpace access info for each Space 
+- Exposes utility functions to get RestConfig by simpy specifying the space name  
+
+Currently the SAC exposes a simple function that supply the config access info for a space. In the future we will add a SAC space aware ClientSets
+
+## Architecture details 
+When a new SpaceProviderDesc is created the SM creates a new namespace that will be used to host all the Space objects representing pSpaces on that space provider.  
```

Comment:
But there can be more than one SpaceProviderDesc for a given provider (ProviderType), right?

Reply:
Yes. But the above say it's a NS per provider (instance) not for provider type.

Reply:
If "SpaceProviderDesc" is synonymous with "space provider" then you have answered the question but invalidated the earlier statement that KIND is an example of a space provider.

Reply:
SpaceProviderDesc is the object/CRD that holds information about the space provider itself.

Reply:
In that case some other text is wrong, the text that says KIND is an example of a space provider. In fact KIND is an example of a SpaceProviderType, _not_ a provider (SpaceProviderDesc).

Reply:
There can be two SpaceProviderDesc with the same SpaceProviderType, right?

Reply:
Yes. However, The assumption is that we will have one SpaceProviderDesc per space provider instance. So you can have to KCP servers running and create 2 SpaceProviderDesc objects to represent them . Both will be with the same "provider type" of KCP. 
In theory , nothing prohibit you to have 2 SpaceProviderDesc  pointing to the SAME space provider instance, there is a potential of management collision  when a single space is being managed through 2 different SpaceProviderDesc  and basically represented as 2 different Spaces

Reply:
I was not asking about the case of two SpaceProviderDesc referring to the same provider, I was asking about the case of two SpaceProviderDesc referring to two providers of the same _type_. Since you seem to be confirming that this is allowed, that means you have a conceptual mismatch in the text that says that "kcp" --- which the reader will interpret as referring to the product from the kcp repo --- is an example of a provider (i.e., SpaceProviderDesc); instead "kcp" (the product) is an example of a SpaceProviderType, and multiple SpaceProviderDesc may refer to the same product.

Reply:
I'm not sure I understand the issue. For the space provider we named the "providerType" in the same name as the name of the product which is also the actual provider backend. Maybe I should have mentioned that there can be multiple provider backend instances represented by multiple providerDesc  and they all of type KCP

Reply:
My point is that this design has one extra level of multiplicity that is not properly introduced and consistently accounted for, it only appears in passing in some places. The English text often talks about the concept of a "space provider", and the Terminology section defines that term and says that "KIND" is an example of a "space provider". But in the places where the precision of code is used, there are *two* concepts --- SpaceProviderDesc and SpaceProviderType --- and they are different. "KIND" is an example of a SpaceProviderType, and yet you also want to treat SpaceProviderDesc as equivalent to "space provider".

Reply:
Will review the doc again. I agree that for space we have distinct pSpace Vs Space, where on the backend we have space provider, space provider object (SpaceProviderDesc) and space provider adaptor (and space provider type)